File: backend/src/main/java/iuh/labbooking/service/statistics/LabBookingStatisticsService.java
package iuh.labbooking.service.statistics;

import iuh.labbooking.dto.response.base.PageResponse;
import iuh.labbooking.dto.response.statistics.*;
import iuh.labbooking.enums.RequestStatus;

import java.time.LocalDate;
import java.util.List;

public interface LabBookingStatisticsService {
    LabBookingKpiResponse getKpis(LocalDate startDate, LocalDate endDate);
    List<BookingOutcomeResponse> getOutcomeDistribution(LocalDate startDate, LocalDate endDate);
    List<SubmissionTrendResponse> getSubmissionTrend(LocalDate startDate, LocalDate endDate);
    PageResponse<List<BookingAuditLogResponse>> getAuditLogs(
            LocalDate startDate, LocalDate endDate, RequestStatus status, String adminId, int page, int limit);
}

File: frontend/src/services/lab-booking-statistics.service.ts
import http from '@/lib/http'
import type { ApiResponse, PageResponse } from '@/schemas/base.schema'
import type {
  AuditLogParams,
  BookingAuditLogResponse,
  BookingOutcomeResponse,
  LabBookingKpiParams,
  LabBookingKpiResponse,
  SubmissionTrendResponse
} from '@/schemas/lab-booking-statistics.schema'

const labBookingStatisticsService = {
  getKpis: (params: LabBookingKpiParams) =>
    http.get<ApiResponse<LabBookingKpiResponse>>('/admin/statistics/lab-bookings/kpis', { params }),

  getOutcomeDistribution: (params: LabBookingKpiParams) =>
    http.get<ApiResponse<BookingOutcomeResponse[]>>('/admin/statistics/lab-bookings/outcome-distribution', {
      params
    }),

  getSubmissionTrend: (params: LabBookingKpiParams) =>
    http.get<ApiResponse<SubmissionTrendResponse[]>>('/admin/statistics/lab-bookings/submission-trend', {
      params
    }),

  getAuditLogs: (params: AuditLogParams) =>
    http.get<ApiResponse<PageResponse<BookingAuditLogResponse[]>>>('/admin/statistics/lab-bookings/audit-logs', {
      params
    })
}

export default labBookingStatisticsService
